keanu-project/src/test/java/io/improbable/keanu/vertices/intgr/probabilistic/FuzzyCastToIntegerVertexTest.java

```diff
@@ -153,16 +153,16 @@ public void calculate_dP_dmu() {
         fuzzy.setValue(observedValue);
 
         mu.setValue(mu1);
-        double density1 = fuzzy.densityAtValue();
-        double actual_dPdmu = fuzzy.dDensityAtValue().get(mu.getId());
+        double logDensity1 = fuzzy.logProbAtValue();
+        double actual_dPdmu = fuzzy.dLogProbAtValue().get(mu.getId());
 
         mu.setValue(mu2);
-        double density2 = fuzzy.densityAtValue();
+        double logDensity2 = fuzzy.logProbAtValue();
 
-        double expected_dPdmu = (density2 - density1) / delta;
+        double expected_dPdmu = (logDensity2 - logDensity1) / delta;
 
         log.info("Expected = " + expected_dPdmu + ", Actual = " + actual_dPdmu);
-        assertEquals(expected_dPdmu, actual_dPdmu, 1e-5);
+        assertEquals(expected_dPdmu, actual_dPdmu, 1e-4);
```

Comment:
Was the test failing at e-5?

Reply:
Yeah, the error from using the logProb is slightly higher. I think this is just due to the logProb being a bit steeper at this point than the density was. It can also be fixed by decreasing the size of delta.

Reply:
Nice, just confirming